docs/IMPLEMENTATION.md

```diff
@@ -0,0 +1,154 @@
+```
+bundle update faraday --conservative 
+```
+Make sure you have successfully bundled and resolved any dependency issues before moving onto the generator commands.
+
+
+## Step 2
+Run the install generator command in your terminal.
+
+Use the `--datacite` flag if working with DataCite DOIs:
+```
+rails g hyrax:doi:install --datacite
+```
+If not using DataCite DOIs:
+```
+rails g hyrax:doi:install
+```
+
+## Step 3
+
+Run the generator to add DOI support to a given work type.
+
+Add the `--datacite` flag if creating DataCite DOIs:
+```
+rails g hyrax:doi:add_to_work_type MyWorkType --datacite
+```
+If not using DataCite DOIs:
+
+```
+rails g hyrax:doi:add_to_work_type MyWorkType
+```
+
+* This will add 2 properties to your worktype's model: `doi` and `doi_status_when_public`  At this point you could load your app and see the new DOI fields on the edit/new form for work types but there is currently no connection to the DOI Registration Agent (RA).
+* It will also add the DOI behavior to the worktype's presenter and you'll see the value of that in Step 6.
+* Worth noting that if you are working with DataCite DOIs and your work types do not have `publisher` metadata you'll need to add those to the form for the autofill button to be able to drop the value of publisher from DataCite into that field.
+
+## Step 4
+Set up connection to DOI Registration Agent.  
+* Get credentials from RA of choice - in this example we will use [DataCite](https://datacite.org/) and their test site [Fabrica](https://doi.test.datacite.org/).
+* Obtain testing credentials from your RA administrator or request direct from support@datacite.org.
+* Obtain production credentials from your RA administrator.
+* Add the credentials to your application's `.env` or where ever your credentials are stored.  Note that if that place is not in your `.env` file then be sure to update the location added by the generate command earlier in `config/initializers/hyrax-doi.rb` by default it looks like this if you generated with the `--datacite` flag.
+```
+Hyrax::DOI::DataCiteRegistrar.prefix = ENV['DATACITE_PREFIX']
+Hyrax::DOI::DataCiteRegistrar.username = ENV['DATACITE_USERNAME']
+Hyrax::DOI::DataCiteRegistrar.password = ENV['DATACITE_PASSWORD']
+```
+* Should you be using something like a Helm Chart to manage your production and development environment variables then you will want to store your production credentials in the `app_name/chart/production-values.yaml.enc` and your test credentials from Fabrica in the `app_name/chart/dev-values.yaml.enc`
+
+## Step 5
+Connect your application's metadata to the metadata provided by your DOI RA.  See this wiki page on [Crosswalking Metadata with Bolognese](https://github.com/K8Sewell/hyrax-doi/wiki/Metadata-Crosswalking-with-Bolognese) for additional details.
```

Comment:
Is this supposed to link to your fork Kait?

Reply:
Nope!  Thanks for the catch!